module/zfs/arc.c

```diff
@@ -4882,31 +4882,6 @@ arc_kmem_reap_soon(void)
 static boolean_t
 arc_evict_cb_check(void *arg, zthr_t *zthr)
 {
-	/*
-	 * This is necessary so that any changes which may have been made to
-	 * many of the zfs_arc_* module parameters will be propagated to
-	 * their actual internal variable counterparts. Without this,
-	 * changing those module params at runtime would have no effect.
-	 */
-	arc_tuning_update(B_FALSE);
-
-	/*
-	 * This is necessary in order to keep the kstat information
-	 * up to date for tools that display kstat data such as the
-	 * mdb ::arc dcmd and the Linux crash utility.  These tools
-	 * typically do not call kstat's update function, but simply
-	 * dump out stats from the most recent update.  Without
-	 * this call, these commands may show stale stats for the
-	 * anon, mru, mru_ghost, mfu, and mfu_ghost lists. Even
-	 * with this change, the data might be up to 1 second
-	 * out of date(the arc_evict_zthr has a maximum sleep
-	 * time of 1 second); but that should suffice.  The
-	 * arc_state_t structures can be queried directly if more
-	 * accurate information is needed.
-	 */
-	if (arc_ksp != NULL)
-		arc_ksp->ks_update(arc_ksp, KSTAT_READ);
```

Comment:
>  its usefulness does not outweigh its negative impact on performance.

Can you quantify its impact on performance?  When this was added it was determined to be negligible.  If that's not the case perhaps we could instead wrap this chunk in `ZFS_DEBUG` so it's only enabled in debug builds.

Reply:
@mjguzik can you describe the performance problem here?
I can wrap it in `ZFS_DEBUG`

Reply:
I'll quote some parts of the original conversation I had with @mjguzik in his absence:

> [...] called very second and has grand time doing a lot of work typically for no reason
> [...] 'crash' is advanced enough that it can be made to sum it up on its own
> [...] the call here only keeps them up to date for the purpose of reading them in the debugger
> which it should be able to do on its own
```
dtrace -n 'fbt::aggsum_value:entry { @[execname] = count(); }'
sysctl kstat.zfs.misc.arcstats | head
zfskern                                                          26
sysctl                                                          312
```
> [...] general thing here is that zfs on an idle system is having an atomic party

He mentioned aggsum, which we do know is inefficient.